plutus-benchmark/bitwise/src/PlutusBenchmark/Ed25519/Compiled.hs

```diff
@@ -0,0 +1,35 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+module PlutusBenchmark.Ed25519.Compiled (
+  checkValidCompiled,
+  msgAsData,
+  signatureAsData,
+  pkAsData
+  ) where
+
+import PlutusBenchmark.Ed25519 (checkValid)
+import PlutusTx.Code (CompiledCode)
+import PlutusTx.IsData (toBuiltinData, unsafeFromBuiltinData)
+import PlutusTx.Plugin ()
+import PlutusTx.Prelude
+import PlutusTx.TH (compile)
+
+checkValidCompiled :: CompiledCode (BuiltinData -> BuiltinData -> BuiltinData -> Bool)
+checkValidCompiled = $$(compile [|| \signature msg pk -> checkValid (unsafeFromBuiltinData signature)
+                                                                    (unsafeFromBuiltinData msg)
+                                                                    (unsafeFromBuiltinData pk) ||])
+
+msgAsData :: CompiledCode BuiltinData
+msgAsData = $$(compile [|| toBuiltinData ("hello world" :: BuiltinByteString) ||])
```

Comment:
`BuiltinData` can be lifted directly into `CompiledCode`:

```haskell
msgAsData :: CompiledCode BuiltinData
msgAsData = liftCodeDef (toBuiltinData ("hello world" :: BuiltinByteString))
```

Reply:
This replacement appears fragile. This change here:

```haskell
signatureAsData :: CompiledCode BuiltinData
signatureAsData = liftCodeDef (toBuiltinData ("\NUL\147!x\173\167\209z`\t\243|\195$X$\233\166\234\NUL\134\152l\DC4\243\&4\217\NAK\152\180{$M\227R\214\218%\241\157\ENQ\SO\ENQ\t\152\140\171\240\200f\184\133\203\227z\163\NUL\185\155Y\139\178\249\STX" :: BuiltinByteString))

{-
signatureAsData :: CompiledCode BuiltinData
signatureAsData =
  $$(compile [|| toBuiltinData ("\NUL\147!x\173\167\209z`\t\243|\195$X$\233\166\234\NUL\134\152l\DC4\243\&4\217\NAK\152\180{$M\227R\214\218%\241\157\ENQ\SO\ENQ\t\152\140\171\240\200f\184\133\203\227z\163\NUL\185\155Y\139\178\249\STX" :: BuiltinByteString) ||])
-}
```

produces this golden failure:

```
    test
      9.6
        Ed25519:         FAIL (0.01s)
          Test output was different from 'bitwise/test/9.6/Ed25519.pir.golden'. Output of ["diff","-u","bitwise/test/9.6/Ed25519.pir.golden","/tmp/nix-shell.YeLs4n/Ed25519.pir341253-4.actual"]:
          --- bitwise/test/9.6/Ed25519.pir.golden	2024-08-06 10:37:15.782140753 +1200
          +++ /tmp/nix-shell.YeLs4n/Ed25519.pir341253-4.actual	2024-08-06 10:43:20.606177116 +1200
          @@ -2319,6 +2319,6 @@
                           (/\dead -> ifThenElse {Bool} (equalsInteger y y) True False)
                           (/\dead -> False)
                           {all dead. dead})))
          -  (B #c080c2932178c2adc2a7c3917a6009c3b37cc383245824c3a9c2a6c3aac080c286c2986c14c3b334c39915c298c2b47b244dc3a352c396c39a25c3b1c29d050e0509c298c28cc2abc3b0c38866c2b8c285c38bc3a37ac2a3c080c2b9c29b59c28bc2b2c3b902)
          +  (B #00c2932178c2adc2a7c3917a6009c3b37cc383245824c3a9c2a6c3aa00c286c2986c14c3b334c39915c298c2b47b244dc3a352c396c39a25c3b1c29d050e0509c298c28cc2abc3b0c38866c2b8c285c38bc3a37ac2a300c2b9c29b59c28bc2b2c3b902)
             (B #68656c6c6f20776f726c64)
             (B #283ac3bfc3bbc281372d5e77c3bdc2910b681bc2ab72c2bdc39f2fc395517a62c3b9c2af247ac39371c38311c386)
          \ No newline at end of file
          
          Use -p '$0=="bitwise.Ed25519.test.9.6.Ed25519"' to rerun this test only.
        Ed25519:         OK
        Ed25519:         FAIL (0.02s)
          Test output was different from 'bitwise/test/9.6/Ed25519.budget.golden'. Output of ["diff","-u","bitwise/test/9.6/Ed25519.budget.golden","/tmp/nix-shell.YeLs4n/Ed25519.budget341253-6.actual"]:
          --- bitwise/test/9.6/Ed25519.budget.golden	2024-08-06 10:37:18.085499655 +1200
          +++ /tmp/nix-shell.YeLs4n/Ed25519.budget341253-6.actual	2024-08-06 10:43:20.629510707 +1200
          @@ -1,2 +1,10 @@
          -({cpu: 2137056248968
          -| mem: 7088454370})
          \ No newline at end of file
          +An error has occurred:
          +The machine terminated because of an error, either from a built-in function or from an explicit use of 'error'.
          +Caused by: integerToByteString
          +             False
          +             32
          +             -705236757155852610088934725679574536026126424420972475413373104382334263805059295045975968846120247953652472732782828952697926196694920784559513127758387
          +Final budget: ({cpu: 7896928821
          +| mem: 26930816})
          +Logs: integerToByteString: cannot convert negative Integer
          +Input: -705236757155852610088934725679574536026126424420972475413373104382334263805059295045975968846120247953652472732782828952697926196694920784559513127758387
          \ No newline at end of file
          
          Use -p '$0=="bitwise.Ed25519.test.9.6.Ed25519"' to rerun this test only.
        Ed25519:         FAIL (0.01s)
          Test output was different from 'bitwise/test/9.6/Ed25519.eval.golden'. Output of ["diff","-u","bitwise/test/9.6/Ed25519.eval.golden","/tmp/nix-shell.YeLs4n/Ed25519.eval341253-7.actual"]:
          --- bitwise/test/9.6/Ed25519.eval.golden	2024-08-06 10:37:20.305524297 +1200
          +++ /tmp/nix-shell.YeLs4n/Ed25519.eval341253-7.actual	2024-08-06 10:43:20.646177556 +1200
          @@ -1 +1,6 @@
          -(constr 1)
          \ No newline at end of file
          +An error has occurred:
          +The machine terminated because of an error, either from a built-in function or from an explicit use of 'error'.
          +Caused by: integerToByteString
          +             False
          +             32
          +             -705236757155852610088934725679574536026126424420972475413373104382334263805059295045975968846120247953652472732782828952697926196694920784559513127758387
          \ No newline at end of file
          
          Use -p '$0=="bitwise.Ed25519.test.9.6.Ed25519"' to rerun this test only.
```

It is specifically this change and no other. The other lifted arguments work fine.

Reply:
@zliu41^ Do you have any comment on this?  I don't know much about this stuff.